ballerina/SQLClient.bal

```diff
@@ -0,0 +1,181 @@
+        sql:ParameterizedQuery query = sql:queryConcat(`UPDATE `, self.tableName, ` SET`, check self.getSetClauses('object));
+
+        if !(filter is ()) {
+            query = sql:queryConcat(query, ` WHERE`);
+            if filter is FilterQuery {
+                query = sql:queryConcat(query, ` `, filter);
+            } else {
+                query = sql:queryConcat(query, check self.getWhereClauses(filter));
+            }
+        }
+
+        _ = check self.dbClient->execute(query);
+    }
+
+    function runDeleteQuery(map<anydata>|FilterQuery? filter) returns error? {
+        sql:ParameterizedQuery query = sql:queryConcat(`DELETE FROM `, self.tableName);
+
+        if !(filter is ()) {
+            query = sql:queryConcat(query, ` WHERE`);
+            if filter is FilterQuery {
+                query = sql:queryConcat(query, ` `, filter);
+            } else {
+                query = sql:queryConcat(query, check self.getWhereClauses(filter));
+            }
+        }
+
+        _ = check self.dbClient->execute(query);
+    }
+
+    private function getInsertQueryParams(record {} 'object) returns sql:ParameterizedQuery {
+        sql:ParameterizedQuery params = `(`;
+        string[] keys = self.fieldMetadata.keys();
+        int columnCount = 0;
+        foreach string key in keys {
+            if self.fieldMetadata.get(key).autoGenerated {
+                continue;
+            }
+            if columnCount > 0 {
+                params = sql:queryConcat(params, `,`);
+            }
+            params = sql:queryConcat(params, `${<sql:Value>'object[key]}`);
+            columnCount = columnCount + 1;
+        }
+        params = sql:queryConcat(params, `)`);
+        return params;
+    }
+
+    private function getColumnNames() returns sql:ParameterizedQuery {
+        sql:ParameterizedQuery params = ` `;
+        string[] keys = self.fieldMetadata.keys();
+        int columnCount = 0;
+        foreach string key in keys {
+            if self.fieldMetadata.get(key).autoGenerated {
+                continue;
+            }
+            if columnCount > 0 {
+                params = sql:queryConcat(params, `, `);
+            }
+            params = sql:queryConcat(params, stringToParameterizedQuery(self.fieldMetadata.get(key).columnName));
+            columnCount = columnCount + 1;
+        }
+        return params;
+    }
+
+    // TODO: handle composite keys (record types)
+    private function getGetKeyWhereClauses(anydata key) returns sql:ParameterizedQuery|error {
+        map<anydata> filter = {};
+        filter[self.keyFields[0]] = key;
+        return check self.getWhereClauses(filter);
+    }
+
+    function getWhereClauses(map<anydata> filter) returns sql:ParameterizedQuery|error {
+        sql:ParameterizedQuery query = ` `;
+    
+        string[] keys = filter.keys();
+        foreach int i in 0 ..< keys.length(){
```

Comment:
```suggestion
        foreach int i in 0 ..< keys.length() {
```
Shall we format the file?

Reply:
[420ee3c](https://github.com/ballerina-platform/module-ballerina-persist/pull/1/commits/420ee3c3b7c3095f7d4e87d3ca0af9eb0d961c9e)